snuba/admin/static/tracing/index.tsx

```diff
@@ -1,8 +1,115 @@
-import React from "react";
+import React, { useState } from "react";
 import Client from "../api_client";
 import QueryDisplay from "../components/query_display/query_display";
 import { QueryResult } from "../components/query_display/types";
 
+type LogLine = {
+  host: string;
+  pid: string;
+  query_id: string;
+  log_level: string;
+  component: string;
+  message: string;
+};
+
+type BucketedLogs = Map<String, Map<MessageCategory, LogLine[]>>;
+
+const logLineMatcher =
+  /\[ (?<hostname>\S+) \] \[ (?<pid>\d+) \] \{(?<local_query_id>[^}]+)\} <(?<log_level>[^>]+)> (?<component>[^:]+): (?<message>.*)/;
+
+function parseLogLine(logLine: string): LogLine | null {
+  const logLineRegexMatch = logLine.match(logLineMatcher);
+  const host = logLineRegexMatch?.groups?.hostname;
+  const pid = logLineRegexMatch?.groups?.pid;
+  const local_query_id = logLineRegexMatch?.groups?.local_query_id;
+  const log_level = logLineRegexMatch?.groups?.log_level;
+  const message = logLineRegexMatch?.groups?.message;
+  const component = logLineRegexMatch?.groups?.component;
+
+  if (host && pid && local_query_id && log_level && message) {
+    const logLineParsed: LogLine = {
+      host: host,
+      pid: pid!,
+      query_id: local_query_id,
+      log_level: log_level!,
+      component: component!,
+      message: message!,
+    };
+    return logLineParsed;
+  } else {
+    return null;
+  }
+}
+
+enum MessageCategory {
+  housekeeping,
+  select_execution,
+  aggregation,
+  memory_tracker,
+  unknown,
+}
+
+function getMessageCategory(logLine: LogLine): MessageCategory {
+  const component = logLine.component;
+  if (
+    component.match(/^ContextAccess|AccessRightsContext|^executeQuery/) &&
+    !logLine.message.startsWith("Read")
+  ) {
+    return MessageCategory.housekeeping;
+  } else if (
+    component.match(
+      /^(default.(\S+) \(SelectExecutor\)|MergeTreeSelectProcessor)/
+    )
+  ) {
+    return MessageCategory.select_execution;
+  } else if (
+    component.match(/^InterpreterSelectQuery|AggregatingTransform|Aggregator/)
+  ) {
+    return MessageCategory.aggregation;
+  } else if (
+    component.match(/MemoryTracker/) ||
+    (component.match(/^executeQuery/) && logLine.message.startsWith("Read"))
+  ) {
+    return MessageCategory.memory_tracker;
+  } else {
+    return MessageCategory.unknown;
+  }
+}
+
+let collapsibleStyle = { listStyleType: "none", fontFamily: "Monaco" };
+
+function NodalDisplay(props: {
+  host: string;
+  category: MessageCategory;
+  title?: string;
+  logsBucketed: BucketedLogs;
+}) {
+  const [visible, setVisible] = useState<boolean>(false);
+
+  const nodeKey = props.host + "-" + props.category;
+  return (
+    <li key={nodeKey}>
+      <span onClick={() => setVisible(!visible)}>
+        {props.title} - Click to {visible ? "collapse" : "expand"}
```

Comment:
Could we use a symbol to show this state instead of "Click to X"?

Reply:
Looks like Lyn is creating a new component on another branch -- maybe for now this could be [+] or [-] depending on current visibility state?

Reply:
I like [+] and [-]. Collapse component was more for history, we could try it out but sure if it would look right here.

Reply:
+1 for [+]/[-]